### ISSUE TITLE
Symfony PHP7 specific code

### ISSUE THREAD

**Issue by nevadascout** [HvyIndustries/crane#154]
The version of Symfony on the `master` branch of their github page uses PHP7.1 specific code in part of their test fixtures which the php-parser doesn't seem to understand.

I am using PHP 5.6 and the line in question is being reported as an error by VS Code's php linter: `syntax error, unexpected '?', expecting variable (T_VARIABLE)`

``` php
namespace Symfony\Component\HttpKernel\Tests\Fixtures\Controller;

class NullableController
{
    public function action(?string $foo, ?\stdClass $bar, ?string $baz = 'value', $mandatory)
    {
    }
}
```

Removing the question marks makes the linter then throw a different error: `Default value for parameters with a class type hint can only be NULL` and the parameters are in the wrong order - optional parameters must be at the end of the method definition.

**The php-parser should be updated to support this PHP 7.1 feature**

Generated error in debug log:

```
[TypeError: Cannot read property '0' of undefined]
Issue processing e:\php-frameworks\symfony-master\src\Symfony\Component\HttpKernel\Tests\Fixtures\Controller\NullableController.php
```

_Framework: **Symfony 3.1**_

**Comment by nevadascout** [comment]
One error out of 3,200 files isn't bad!

**Comment by ichiriac** [comment]
this bug is now fixed with the 0.1.0 release of php-parser

**Comment by ichiriac** [comment]
This issue can be closed now, bellow the results of parsing :

> PS : as PHP7 is fully supported, new properties as nullable, variadic or type are now available

PHP Script :
```php
<?php
namespace Symfony\Component\HttpKernel\Tests\Fixtures\Controller;

class NullableController
{
    public function action(?string $foo, ?\stdClass $bar, ?string $baz = 'value', $mandatory)
    {
    }
}
```

AST Structure :
```json
{
  "kind": "program",
  "children": [
    {
      "kind": "namespace",
      "children": [
        {
          "kind": "class",
          "name": "NullableController",
          "isAnonymous": false,
          "extends": null,
          "implements": null,
          "body": [
            {
              "kind": "method",
              "name": "action",
              "arguments": [
                {
                  "kind": "parameter",
                  "name": "foo",
                  "value": null,
                  "type": {
                    "kind": "identifier",
                    "resolution": "uqn",
                    "name": "string"
                  },
                  "byref": false,
                  "variadic": false,
                  "nullable": true
                },
                {
                  "kind": "parameter",
                  "name": "bar",
                  "value": null,
                  "type": {
                    "kind": "identifier",
                    "resolution": "fqn",
                    "name": "\\stdClass"
                  },
                  "byref": false,
                  "variadic": false,
                  "nullable": true
                },
                {
                  "kind": "parameter",
                  "name": "baz",
                  "value": {
                    "kind": "string",
                    "value": "value",
                    "isDoubleQuote": false
                  },
                  "type": {
                    "kind": "identifier",
                    "resolution": "uqn",
                    "name": "string"
                  },
                  "byref": false,
                  "variadic": false,
                  "nullable": true
                },
                {
                  "kind": "parameter",
                  "name": "mandatory",
                  "value": null,
                  "type": null,
                  "byref": false,
                  "variadic": false,
                  "nullable": false
                }
              ],
              "byref": false,
              "type": null,
              "nullable": false,
              "body": {
                "kind": "block",
                "children": []
              },
              "isAbstract": false,
              "isFinal": false,
              "visibility": "public",
              "isStatic": false
            }
          ],
          "isAbstract": false,
          "isFinal": false
        }
      ],
      "name": "Symfony\\Component\\HttpKernel\\Tests\\Fixtures\\Controller",
      "withBrackets": false
    }
  ],
  "errors": []
}
```

**Comment by nevadascout** [comment]
Released in v0.3.0
